pkg/apps/registry/deploylog/rest.go

```diff
@@ -215,11 +219,97 @@ func (r *REST) waitForExistingDeployment(namespace, name string) (*kapi.Replicat
 // view its logs.
 func (r *REST) returnApplicationPodName(target *kapi.ReplicationController) (string, error) {
 	selector := labels.SelectorFromValidatedSet(labels.Set(target.Spec.Selector))
-	sortBy := func(pods []*kapiv1.Pod) sort.Interface { return controller.ByLogging(pods) }
+	sortBy := func(pods []*corev1.Pod) sort.Interface { return controller.ByLogging(pods) }
 
-	firstPod, _, err := kcmdutil.GetFirstPod(r.podClient, target.Namespace, selector.String(), r.timeout, sortBy)
+	firstPod, _, err := GetFirstPod(r.podClient, target.Namespace, selector.String(), r.timeout, sortBy)
 	if err != nil {
-		return "", errors.NewInternalError(err)
+		return "", apierrors.NewInternalError(err)
 	}
 	return firstPod.Name, nil
 }
+
+// GetFirstPod returns a pod matching the namespace and label selector
```

Comment:
@deads2k this is the stuff we need to to copy out from kubectl ? can we have follow up issue where we extract this from kubectl into some kind of library in kube?

Reply:
(also if yes, can you add comment/todo that this was copied out? )